crypto/x509/x509_vfy.c

```diff
@@ -471,16 +471,16 @@ static int check_chain(X509_STORE_CTX *ctx)
         int ret;
 
         x = sk_X509_value(ctx->chain, i);
-        CHECK_CB((ctx->param->flags & X509_V_FLAG_IGNORE_CRITICAL) == 0
-                 && (x->ex_flags & EXFLAG_CRITICAL) != 0,
-                 ctx, x, i, X509_V_ERR_UNHANDLED_CRITICAL_EXTENSION);
         CHECK_CB(!allow_proxy_certs && (x->ex_flags & EXFLAG_PROXY),
                  ctx, x, i, X509_V_ERR_PROXY_CERTIFICATES_NOT_ALLOWED);
+
         ret = X509_check_ca(x);
         switch (must_be_ca) {
         case -1:
             CHECK_CB((ctx->param->flags & X509_V_FLAG_X509_STRICT) != 0
-                         && ret != 1 && ret != 0,
+                     && ret != 1 && ret != 0 && ret != 4
+                     /* accept self-signed v1 cert in root position as CA: */
```

Comment:
The "root position" comment is confusing when you actually mean leaf position.

Reply:
Yes, this accepts self-signed EE certs that are not CAs in strict mode. In cases where the peer is explicitly pinned to a particular non-CA EE cert, presumably there are actually no trust-anchors configured, and one is basically just validating a pinned public key.  It is not clear to me why the "strict" flag would be set here?  What is the use-case motivating this change?

Reply:
Right, and actually the position does not matter at this point!
So I've correct the new exception to:
```
                         && ret != 3 /* accept self-signed v1 cert as CA */
```

Reply:
Can you use define's instead of 1, 3,4, etc magic numbers?

Reply:
oops, I see you're already planning on doing that :)